kustomize/move2kube/README.md

```diff
@@ -0,0 +1,78 @@
+## Prerequisites
+* The manifests are generated without the `namespace` configuration
+* Manifest names reflect the `resources` section of [kustomization.yaml](./base/kustomization.yaml)
+
+A brief explanation of the system architecture is provided in the related [document](./move2kube.md).
+
+## Kustomization options
+### Image update
+Run the following to update the default images to a custom configuration
+```
+cd base && kustomize edit set image serverless-workflow-move2kube=quay.io/orchestrator/serverless-workflow-move2kube:1234 && cd ..
+cd base && kustomize edit set image serverless-workflow-m2k-kfunc=quay.io/orchestrator/serverless-workflow-m2k-kfunc:1234 && cd ..
+```
+
+## Configure properties
+Edit configuration in [replace-config.yaml](./base/replace-config.yaml), [config.properties](./base/config.properties) and 
+[secret.properties](./base/secret.properties) to match your environment configuration.
+
+## Deploy to the cluster
+### Deploy the production environment
+This environment applies the generated manifests with minimal customizations to:
+* Use the latest application images
+* Force the `prod` profile for the SonataFlow instance
+* Deploy by default on the `sonataflow-infra` namespace
+* Mount the configurations defined in [Configure properties](#configure-properties) as environment variables
+
+The default namespace can be customized with:
+```bash
+TARGET_NS=YOUR-NS
+cd base && kustomize edit set namespace=$TARGET_NS && cd ..
+```
+
+Once the configuration is set, apply the deployment to the configured namespace with:
+```bash
+kustomize build base | oc apply -f -
```

Comment:
```
$ kustomize build base | oc apply -f -
Error: error in remove for path: '/metadata/creationTimestamp': Unable to remove nonexistent key: creationTimestamp: missing value
error: no objects passed to apply
```

Reply:
It seems the remove is not needed, only CM have the creationTimestamp set to null after the patch (at least that is the output I have)

Reply:
```
$ kustomize build base | oc apply -f -
configmap/01-m2k-resources created
configmap/m2k-props created
configmap/move2kube-config created
configmap/move2kube-values created
secret/move2kube-secret created
Warning: Kubernetes default value is insecure, Knative may default this to secure in a future release: spec.template.spec.containers[0].securityContext.allowPrivilegeEscalation, spec.template.spec.containers[0].securityContext.capabilities, spec.template.spec.containers[0].securityContext.runAsNonRoot, spec.template.spec.containers[0].securityContext.seccompProfile, spec.template.spec.initContainers[0].securityContext.capabilities, spec.template.spec.initContainers[0].securityContext.runAsNonRoot, spec.template.spec.initContainers[0].securityContext.seccompProfile
service/move2kube created
Warning: would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (container "move2kube" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "move2kube" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "move2kube" must set securityContext.runAsNonRoot=true), runAsUser=0 (container "move2kube" must not set runAsUser=0), seccompProfile (pod or container "move2kube" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
deployment.apps/move2kube created
broker.eventing.knative.dev/default created
trigger.eventing.knative.dev/error-trigger-move2kube created
trigger.eventing.knative.dev/m2k-save-transformation-event created
trigger.eventing.knative.dev/transformation-saved-trigger-move2kube created
route.route.openshift.io/move2kube created
sonataflow.sonataflow.org/m2k created
Error from server (BadRequest): error when creating "STDIN": admission webhook "validation.webhook.serving.knative.dev" denied the request: validation failed: must not set the field(s): spec.template.spec.initContainers
pod spec support for init-containers is off, but found 1 init containers: 
```
Yes, for M2K, you need to run the following before:
```
 $ oc patch configmap/config-features \
    -n knative-serving \
    --type merge \
    -p '{"data":{"kubernetes.podspec-init-containers": "enabled", "kubernetes.podspec-securitycontext": "enabled"}}'

$ oc -n sonataflow-infra adm policy add-scc-to-user anyuid -z default
```

Reply:
About the CRD: the issue is that sometimes the transformers convert the `null` timestamp in a `"null"` string, which causes the error, so the patch has to remove it. I missed some files in the latest commit, will add them asap.

About the prerequisites: as I said before, I missed the README changes with a link to the Helm's installation document.

Reply:
I still have an error:
```
$ kustomize build base | oc apply -f -
Error: error in remove for path: '/metadata/creationTimestamp': Unable to remove nonexistent key: creationTimestamp: missing value
error: no objects passed to apply
```